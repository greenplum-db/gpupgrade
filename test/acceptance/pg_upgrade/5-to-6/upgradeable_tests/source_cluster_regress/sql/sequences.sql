-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

-- Test to ensure that sequences can be upgraded.

--------------------------------------------------------------------------------
-- Create and setup upgradeable objects
--------------------------------------------------------------------------------

CREATE SEQUENCE serial_seq START 100;
SELECT nextval('serial_seq');

CREATE TABLE tbl_with_sequence(id INT NOT NULL DEFAULT nextval('serial_seq'), t text);
ALTER SEQUENCE serial_seq OWNED BY tbl_with_sequence.id;

INSERT INTO tbl_with_sequence(t) VALUES('test1');
INSERT INTO tbl_with_sequence(t) VALUES('test2');

SELECT * FROM serial_seq;
