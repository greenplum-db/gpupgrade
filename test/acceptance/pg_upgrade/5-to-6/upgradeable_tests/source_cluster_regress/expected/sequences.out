-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

-- Test to ensure that sequences can be upgraded.

--------------------------------------------------------------------------------
-- Create and setup upgradeable objects
--------------------------------------------------------------------------------

CREATE SEQUENCE serial_seq START 100;
CREATE
SELECT nextval('serial_seq');
 nextval 
---------
 100     
(1 row)

CREATE TABLE tbl_with_sequence(id INT NOT NULL DEFAULT nextval('serial_seq'), t text);
CREATE
ALTER SEQUENCE serial_seq OWNED BY tbl_with_sequence.id;
ALTER

INSERT INTO tbl_with_sequence(t) VALUES('test1');
INSERT 1
INSERT INTO tbl_with_sequence(t) VALUES('test2');
INSERT 1

SELECT * FROM serial_seq;
 sequence_name | last_value | increment_by | max_value           | min_value | cache_value | log_cnt | is_cycled | is_called 
---------------+------------+--------------+---------------------+-----------+-------------+---------+-----------+-----------
 serial_seq    | 102        | 1            | 9223372036854775807 | 1         | 1           | 30      | f         | t         
(1 row)
